### ISSUE TITLE
youtube-dl is broken

### ISSUE THREAD

**Issue by jmintb** [dxli94/WLASL#56]
**TLDR**: `youtube-dl` is broken, to download youtube videos switch to [yt-dlp](https://github.com/yt-dlp/yt-dlp), and update `./start_kit/video_downloader.py`.

As fare as I can tell `youtube-dl` does not work well with youtube any longer, `./start_kit/video_downloader.py` fails to download any youtube videos, where as`yt-dlp` works perfectly. Since it is a fork and has essentially the same CLI interface, I could just replace `youtube-dl` with `yt-dlp` in `./start_kit/video_downloader.py` and everything worked.

I am not really sure about the state of the two projects, from a quick glance `yt-dlp` seems to be in a better state and more up to date, see this [issue](https://github.com/ytdl-org/youtube-dl/issues/30568).

I am using the latest versions.
`youtube-dl` version: 2021.12.17
`yt-dlp` version: 2022.09.01

If there is any interest I can submit a PR with the required changes to use`yt-dlp`. it could be made optional if you prefer to support both as the change is quite minimal.

**Comment by dxli94** [comment]
Hi @jmintb ,

Thanks a lot for the information. Your suggestion sounds great. I'd appreciate if you could submit a PR.

Regards.

**Comment by jmintb** [comment]
Sorry got dragged away from this, will submit a PR soon :)

**Comment by rohit-chandra** [comment]
Can you please tell me how to replace youtube-dl with yt-dlp in ./start_kit/video_downloader.py?

**Comment by jmintb** [comment]
> Can you please tell me how to replace youtube-dl with yt-dlp in ./start_kit/video_downloader.py?

See the PR for how the replace yt-dlp. :+1:

**Comment by jmintb** [comment]
Closed with #63
